File: samples/VisualBasic/Northwind/Customer.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Customers")>
   Public Class Customer

      <Column>
      Property Address As String

      <Column>
      Property City As String

      <Column>
      Property CompanyName As String

      <Column>
      Property ContactName As String

      <Column>
      Property ContactTitle As String

      <Column>
      Property Country As String

      <Column(IsPrimaryKey:=True)>
      Property CustomerID As String

      <Column>
      Property Fax As String

      <Column>
      Property Phone As String

      <Column>
      Property PostalCode As String

      <Column>
      Property [Region] As String

      <Association(OtherKey:=NameOf(CustomerCustomerDemo.CustomerID))>
      ReadOnly Property CustomerCustomerDemos As New Collection(Of CustomerCustomerDemo)

      <Association(OtherKey:=NameOf(Order.CustomerID))>
      ReadOnly Property Orders As New Collection(Of Order)

   End Class

End Namespace

File: samples/CSharp/Northwind/Customer.cs
using System;
using System.Collections.ObjectModel;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Customers")]
   public class Customer {

      [Column(IsPrimaryKey = true)]
      public string CustomerID { get; set; }

      [Column]
      public string CompanyName { get; set; }

      [Column]
      public string ContactName { get; set; }

      [Column]
      public string ContactTitle { get; set; }

      [Column]
      public string Address { get; set; }

      [Column]
      public string City { get; set; }

      [Column]
      public string Region { get; set; }

      [Column]
      public string PostalCode { get; set; }

      [Column]
      public string Country { get; set; }

      [Column]
      public string Phone { get; set; }

      [Column]
      public string Fax { get; set; }

      [Association(OtherKey = nameof(CustomerCustomerDemo.CustomerID))]
      public Collection<CustomerCustomerDemo> CustomerCustomerDemos { get; } = new Collection<CustomerCustomerDemo>();

      [Association(OtherKey = nameof(Order.CustomerID))]
      public Collection<Order> Orders { get; } = new Collection<Order>();
   }
}
